mindsdb/integrations/handlers/huggingface_api_handler/huggingface_api_handler.py

```diff
@@ -16,118 +17,166 @@ class HuggingFaceInferenceAPIHandler(BaseMLEngine):
 
     name = 'huggingface_api'
 
+    @staticmethod
+    def create_validation(target, args=None, **kwargs):
+
+        if 'using' in args:
+            args = args['using']
+
+        hf_api = HfApi()
+
+        if 'model_name' not in args:
+            # detect model by task
+            task = args.get('task')
+            if task is None:
+                raise Exception('model_name or task have to be specified')
+
+            # get from task
```

Comment:
I don't believe this is necessary, because if the model is not passed, the package itself will take the model defined in the configuration file for that particular task. What do you think?

Reply:
Sorry @ea-rus, my bad. What I meant is this,
```
            config = ConfigParser()
            model_name = config.get_config_dict()['TASK_MODEL_MAP'].get(task)
            if model_name is None:
                raise Exception(f'Model not found by task: {task}')
```
If the model is not specified, the package will use the configuration file to access the model for the relevant task.

Reply:
I thought it is better to detect model name, store it and work only with it in 'predict' method to not make this method more compex.